esphome/core/scheduler.h

```diff
@@ -45,6 +56,18 @@ class Scheduler {
     }
 
     static bool cmp(const std::unique_ptr<SchedulerItem> &a, const std::unique_ptr<SchedulerItem> &b);
+    const char *get_type_str() {
+      switch (this->type) {
+        case SchedulerItem::INTERVAL:
+          return "interval";
+        case SchedulerItem::RETRY:
+          return "retry";
+        case SchedulerItem::TIMEOUT:
+          return "timeout";
+        default:
+          return "INVALID TYPE!";
```

Comment:
This is a waste of memory for a string that should never get used.
```suggestion
          return "";
```

Reply:
Actually, all users of this function are inside `#ifdef ESPHOME_DEBUG_SCHEDULER`, so we can wrap this function inside that condition as well.